nova_vm/src/ecmascript/builtins/indexed_collections/typed_array_objects/typed_array_intrinsic_object.rs

```diff
@@ -366,34 +377,108 @@ impl TypedArrayPrototype {
         todo!()
     }
 
+    /// ### [23.2.3.2 get %TypedArray%.prototype.buffer](https://tc39.es/ecma262/#sec-get-%typedarray%.prototype.buffer)
+    ///
+    /// %TypedArray%.prototype.buffer is an accessor property whose set accessor
+    /// function is undefined.
     fn get_buffer(
-        _agent: &mut Agent,
+        agent: &mut Agent,
         _gc: GcScope<'_, '_>,
 
-        _this_value: Value,
+        this_value: Value,
         _: ArgumentsList,
     ) -> JsResult<Value> {
-        todo!()
+        // 1. Let O be the this value.
+        // 2. Perform ? RequireInternalSlot(O, [[TypedArrayName]]).
+        // 3. Assert: O has a [[ViewedArrayBuffer]] internal slot.
+        // 4. Let buffer be O.[[ViewedArrayBuffer]].
+        let o = require_internal_slot_typed_array(agent, this_value)?;
+
+        // 5. Return buffer.
+        Ok(o.get_viewed_array_buffer(agent).into_value())
     }
 
+    /// ### [23.2.3.3 get %TypedArray%.prototype.byteLength](https://tc39.es/ecma262/#sec-get-%typedarray%.prototype.bytelength)
+    ///
+    /// %TypedArray%.prototype.byteLength is an accessor property whose set
+    /// accessor function is undefined.
     fn get_byte_length(
-        _agent: &mut Agent,
+        agent: &mut Agent,
         _gc: GcScope<'_, '_>,
 
-        _this_value: Value,
+        this_value: Value,
         _: ArgumentsList,
     ) -> JsResult<Value> {
-        todo!()
-    }
-
+        // 1. Let O be the this value.
+        // 2. Perform ? RequireInternalSlot(O, [[TypedArrayName]]).
+        // 3. Assert: O has a [[ViewedArrayBuffer]] internal slot.
+        let o = require_internal_slot_typed_array(agent, this_value)?;
+
+        // 4. Let taRecord be MakeTypedArrayWithBufferWitnessRecord(O, seq-cst).
+        let ta_record = make_typed_array_with_buffer_witness_record(agent, o, Ordering::SeqCst);
+
+        // 5. Let size be TypedArrayByteLength(taRecord).
+        let size = match o {
+            TypedArray::Int8Array(_) => typed_array_byte_length::<i8>(agent, &ta_record),
+            TypedArray::Uint8Array(_) => typed_array_byte_length::<u8>(agent, &ta_record),
+            TypedArray::Uint8ClampedArray(_) => {
```

Comment:
**nitpick:** Could maybe just combine this with `Uint8Array` path since it doesn't affect the byte length. Though I guess with the same argument we could just combine to 3 matches that depend on the byte width.

Reply:
I'll combine types with the same byte width, e.g. the `u/i/f` `*8`, `*16`, `*32` and `*64`.